test/blackbox-tests/test-cases/ctypes/lib-no-modules-field.t

```diff
@@ -0,0 +1,52 @@
+Checking the validation of ctypes with no (modules) field works.
+
+  $ cat > dune-project <<EOF
+  > (lang dune 3.20)
+  > (using ctypes 0.3)
+  > EOF
+
+  $ mkdir bindings
+  $ cat > bindings/dune <<EOF
+  > (library (name mybindings))
+  > EOF
+
+  $ cat > bindings/ffi_bindings.ml <<EOF
+  > module Types (F : Ctypes.TYPE) = struct end
+  > module Functions (F : Ctypes.FOREIGN) = struct end
+  > EOF
+
+  $ cat > dune <<EOF
+  > (library
+  >  (name mylib)
+  >  (libraries mybindings)
+  >  (ctypes
+  >   (external_library_name foo)
+  >   (type_description (instance Types) (functor Ffi_bindings))
+  >   (function_description (instance Functions) (functor Ffi_bindings))
+  >   (generated_entry_point C)))
+  > EOF
+
+  $ dune build
```

Comment:
Could you add a plain-text explanation what is expected here and what is failing? That would make it easier to understand the failure case and expected behavior without knowing the entire context.

Reply:
The `(ctypes (type_description (functor ...)))` fields use a module that tells ctypes about the kind of interface to expect for the ffi. Some validation was added to make sure that the module came from the modules of the stanza in https://github.com/ocaml/dune/pull/12124. However this assumption is incorrect as there are people using this with this particular interface module coming from a separate location. More information can be found in https://github.com/ocaml/dune/issues/13001.